examples/asyncapi-from-object/index.ts

```diff
@@ -0,0 +1,38 @@
+import { TypeScriptGenerator } from '../../src';
+
+const generator = new TypeScriptGenerator({ modelType: 'interface' });
+const AsyncAPIDocument = {
+  asyncapi: '2.1.0',
+  info: {
+    title: 'example',
+    version: '0.1.0'
+  },
+  channels: {
+    '/test': {
+      subscribe: {
+        message: {
+          payload: {
+            $schema: 'http://json-schema.org/draft-07/schema#',
+            type: 'object',
+            additionalProperties: false,
+            properties: {
+              email: {
+                type: 'string',
+                format: 'email'
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+};
+
+export async function generate(logCallback: (msg: string) => void): Promise<void> {
+  const models = await generator.generate(AsyncAPIDocument);
+  for (const model of models) {
+    logCallback(model.result);
+  }
+}
+
+generate(console.log);
```

Comment:
I know it may sound strange but you literally caused collateral damage to my brain when I saw it. I needed a minute to figure out "what? how? why?" 😄 

I strongly suggest:
```ts
export async function generate(logCallback: (msg: string) => void): Promise<void> {
  const models = await generator.generate(AsyncAPIDocument);
  for (const model of models) {
    console.log(model);
  }
}

generate(model.result);
```

Reply:
@derberg now you made my brain explode, can you clarify your example? 😆 

I guess you want the remove all of `logCallback` altogether to simplify the example right? If that's the case, it will be very difficult to test the expected output from the generation and in some cases completely impossible. If we can accept that, then yes we can do that.

Reply:
simplify to maximum is my goal always 😄 
then I guess the only way is files and snapshots